train.py

```diff
@@ -995,62 +995,63 @@ def update_old_args(t_args):
     supported_precisions = ['fp16', 'fp32']
     argparser = argparse.ArgumentParser(description="EveryDream2 Training options")
     argparser.add_argument("--config", type=str, required=False, default=None, help="JSON config file to load options from")
-    args, _ = argparser.parse_known_args()
+    args, argv = argparser.parse_known_args()
 
     if args.config is not None:
-        print(f"Loading training config from {args.config}, all other command options will be ignored!")
+        print(f"Loading training config from {args.config}.")
         with open(args.config, 'rt') as f:
-            t_args = argparse.Namespace()
-            t_args.__dict__.update(json.load(f))
-            update_old_args(t_args) # update args to support older configs
-            args = argparser.parse_args(namespace=t_args)
+            args.__dict__.update(json.load(f))
+            update_old_args(args) # update args to support older configs
```

Comment:
With this change `update_old_args` is no longer necessary since the second pass over the arguments will apply the same defaults:

* `shuffle_tags` defaults to `False`,
* `save_full_precision` defaults to `False`,
* `notebook` defaults to `False`,
* `disable_unet_training` defaults to `False`,
* `rated_dataset` defaults to `False`, and
* `rated_dataset_target_dropout_percent` defaults to `50`.

These are the same defaults provided by `update_old_args`. 😄

Reply:
ahha awesome. thanks for taking a look!

Reply:
i removed the function, thanks for the tip